iroh/src/magicsock.rs

```diff
@@ -1406,6 +1476,31 @@ impl MagicSock {
     }
 }
 
+#[derive(Clone, Debug)]
+enum MappedAddr {
+    NodeId(NodeIdMappedAddr),
+    Ip(IpMappedAddr),
+}
+
+impl MappedAddr {
+    fn from_socket_addr(addr: SocketAddr) -> Result<Self> {
```

Comment:
`impl TryFrom<SocketAddr> for MappedAddr` is the rust way...

But with the suggestion above it would be `impl From<SocketAddr> for MappedAddr`.